transport/http/outbound.go

```diff
@@ -147,6 +147,11 @@ func (o *Outbound) Transports() []transport.Transport {
 	return []transport.Transport{o.transport}
 }
 
+// Chooser returns the outbound's peer chooser.
+func (o *Outbound) Chooser() peer.Chooser {
```

Comment:
as opposed to exposing the chooser function and increasing our API exposure for the purpose of tests, can we move the HTTP transport config test into this package?